iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp

```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
+DEFINE_SCHEMA(Guid_FilteringUsingDescriptorsFieldsFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingDescriptorsFieldsFilter)
+    {
+    ECClassCP classA = GetClass("A");
+    BeGuid instanceGuid1 = BeGuid(true);
+    IECInstancePtr instance1 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid1](IECInstanceR instance) {
+        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid)));
+        });
+    BeGuid instanceGuid2 = BeGuid(true);
+    IECInstancePtr instance2 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid2](IECInstanceR instance) {
+        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid)));
+        });
+    KeySetPtr input = KeySet::Create(bvector<IECInstancePtr>{instance1, instance2});
+
+    // create the rule set
+    PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
+    m_locater->AddRuleSet(*rules);
+    ContentRuleP contentRule = new ContentRule("", 1, false);
+    contentRule->AddSpecification(*new ContentInstancesOfSpecificClassesSpecification(1, "", classA->GetFullName(), false, false));
+    rules->AddPresentationRule(*contentRule);
+
+    // validate default
+    ContentDescriptorCPtr descriptor = GetValidatedResponse(m_manager->GetContentDescriptor(AsyncContentDescriptorRequestParams::Create(s_project->GetECDb(), rules->GetRuleSetId(), RulesetVariables(), nullptr, 0, *input)));
+    ASSERT_TRUE(descriptor.IsValid());
+    ContentCPtr content = GetVerifiedContent(*descriptor);
+    ASSERT_TRUE(content.IsValid());
+    ASSERT_EQ(2, content->GetContentSet().GetSize());
+
+    // validate filtered
+    ContentDescriptorPtr ovr = ContentDescriptor::Create(*descriptor);
+    Utf8StringCR fieldFilter = Utf8PrintfString("%s = StrToGuid(\"%s\")", FIELD_NAME(classA, "GuidProp"), instanceGuid2.ToString().c_str());
+    ovr->SetFieldsFilterExpression(fieldFilter);
+    content = GetVerifiedContent(*ovr);
+    ASSERT_TRUE(content.IsValid());
+    ASSERT_EQ(1, content->GetContentSet().GetSize());
+    RulesEngineTestHelpers::ValidateContentSet(bvector<IECInstanceCP>{instance2.get()}, *content);
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_FilteringUsingDescriptorsInstanceFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingDescriptorsInstanceFilter)

```

Comment:
I'd move the test to `Content_Filtering.cpp` and rename to `DescriptorInstanceFilter_FiltersByGuidProperty`.